src/core/reactor.cc

```diff
@@ -1905,6 +1905,13 @@ reactor::open_file_dma(std::string_view nameref, open_flags flags, file_open_opt
                     attr.fsx_extsize = std::min(options.extent_allocation_size_hint,
                                         file_open_options::max_extent_allocation_size_hint);
 
+                    constexpr uint32_t min_extent_size_hint_alignment{128u << 10}; // 128KB
+                    constexpr uint32_t min_hint_alignment_mask{min_extent_size_hint_alignment - 1u};
+
+                    if (attr.fsx_extsize & min_hint_alignment_mask) {
+                        attr.fsx_extsize = align_up<uint32_t>(attr.fsx_extsize, min_extent_size_hint_alignment);
```

Comment:
Doesn't `align_up`  already leaves aligned values as is, so the if above is excessive?

Reply:
Yes, that's true. It is a leftover from testing - in the beginning I added a warning here to check how it behaves.

Reply:
Fixed